rust-install/src/mock/mod.rs

```diff
@@ -9,32 +16,35 @@ pub struct MockInstallerBuilder {
 
 // A component name, the installation commands for installing files
 // (either "file:" or "dir:") and the file paths and contents.
-pub type MockComponent = (&'static str, Vec<MockCommand>, Vec<(&'static str, String)>);
+pub type MockComponent = (String, Vec<MockCommand>, Vec<(String, Vec<u8>)>);
```

Comment:
There are a few places where `MockInstallerBuilder`s are a constructed which haven't been updated to pass in `String`s instead of `&str`s. (See the travis build failures)